simtools/data_model/metadata_collector.py

```diff
@@ -165,85 +175,33 @@ def get_site(self, from_input_meta=False):
-
-        Parameters
-        ----------
-        associated_elements_dict: dict
-            Dictionary for associated elements field.
-
-        Raises
-        ------
-        TypeError, KeyError
-            if error reading association metadata from args.
-        KeyError
-            if metadata description cannot be filled.
-
-        """
-        self._logger.debug(f"Fill metadata from args: {self.args_dict}")
-
-        _association = {}
-
-        try:
-            if self.args_dict and self.args_dict.get("site", None):
-                _association["site"] = names.validate_site_name(self.args_dict["site"])
-            if "telescope" in self.args_dict:
-                _telescope_name = names.validate_array_element_name(self.args_dict["telescope"])
-                _association["class"] = "telescope"
-                _association["type"] = names.get_array_element_type_from_name(_telescope_name)
-                _association["subtype"] = ""
-        except (TypeError, KeyError):
-            self._logger.error("Error reading association metadata from args")
-            raise
-
-        self._fill_context_sim_list(associated_elements_dict, _association)
-
-    def _fill_context_from_input_meta(self, context_dict):
-        """
-        Read and validate input metadata from file and fill CONTEXT metadata fields.
+        Fill context metadata fields with product metadata from input data.
 
         Parameters
         ----------
         context_dict: dict
-            Dictionary with context level metadata.
-
-        Raises
-        ------
-        KeyError
-            if corresponding fields cannot by accessed in the top-level or metadata dictionaries.
+            Dictionary for context metadata fields.
 
         """
-        try:
-            self._merge_config_dicts(context_dict, self.input_metadata[self.observatory]["context"])
-            for key in ("document", "associated_elements", "associated_data"):
-                self._copy_list_type_metadata(
-                    context_dict, self.input_metadata[self.observatory], key
-                )
-        except KeyError:
-            self._logger.debug("No context metadata defined in input metadata file.")
-
-        try:
-            self._fill_context_sim_list(
-                context_dict["associated_data"], self.input_metadata[self.observatory]["product"]
-            )
+        try:  # wide try..except as for some cases we expect that there is no product metadata
+            reduced_product_meta = {
+                key: value
+                for key, value in self.input_metadata[self.observatory]["product"].items()
+                if key in {"description", "id", "creation_time", "valid", "format", "filename"}
+            }
+            self._fill_context_sim_list(context_dict["associated_data"], reduced_product_meta)
         except (KeyError, TypeError):
             self._logger.debug("No input product metadata appended to associated data.")
```

Comment:
> Firstly that using `.get("product")` is safer and deals with `KeyError`. Unless you want to explicitly raise an error to flag up missing metadata

The issue with that solution (which I tried before), is that it doesn't catch the KeyError when `self.input_metadata` does not have the `self.observatory` key (or if input_metadata is set to None). I otherwise agree that get `get` would be the better solution.

On your second point: 

- we copy a subset of the fields from `products` (most importantly is the `PRODUCT:ID`
- the dict is small - see e.g., the examples in the top level metadat documentation. Iteration goes over maybe 10 fields.